go/parquet/internal/testutils/primitive_typed.go

```diff
@@ -0,0 +1,305 @@
+		w.WriteBatchSpaced(p.Values.([]float64), defLevels, repLevels, validBits, validBitsOffset)
+	case *file.Int96ColumnChunkWriter:
+		w.WriteBatchSpaced(p.Values.([]parquet.Int96), defLevels, repLevels, validBits, validBitsOffset)
+	case *file.ByteArrayColumnChunkWriter:
+		w.WriteBatchSpaced(p.Values.([]parquet.ByteArray), defLevels, repLevels, validBits, validBitsOffset)
+	case *file.BooleanColumnChunkWriter:
+		w.WriteBatchSpaced(p.Values.([]bool), defLevels, repLevels, validBits, validBitsOffset)
+	case *file.FixedLenByteArrayColumnChunkWriter:
+		w.WriteBatchSpaced(p.Values.([]parquet.FixedLenByteArray), defLevels, repLevels, validBits, validBitsOffset)
+	default:
+		panic("unimplemented")
+	}
+}
+
+func (p *PrimitiveTypedTest) ReadBatch(reader file.ColumnChunkReader, batch, valuesRead int64, defLevels, repLevels []int16) int64 {
+	switch r := reader.(type) {
+	case *file.Int32ColumnChunkReader:
+		_, read, _ := r.ReadBatch(batch, p.ValuesOut.([]int32)[valuesRead:], defLevels, repLevels)
+		return int64(read)
+	case *file.Int64ColumnChunkReader:
+		_, read, _ := r.ReadBatch(batch, p.ValuesOut.([]int64)[valuesRead:], defLevels, repLevels)
+		return int64(read)
+	case *file.Float32ColumnChunkReader:
+		_, read, _ := r.ReadBatch(batch, p.ValuesOut.([]float32)[valuesRead:], defLevels, repLevels)
+		return int64(read)
+	case *file.Float64ColumnChunkReader:
+		_, read, _ := r.ReadBatch(batch, p.ValuesOut.([]float64)[valuesRead:], defLevels, repLevels)
+		return int64(read)
+	case *file.Int96ColumnChunkReader:
+		_, read, _ := r.ReadBatch(batch, p.ValuesOut.([]parquet.Int96)[valuesRead:], defLevels, repLevels)
+		return int64(read)
+	case *file.ByteArrayColumnChunkReader:
+		_, read, _ := r.ReadBatch(batch, p.ValuesOut.([]parquet.ByteArray)[valuesRead:], defLevels, repLevels)
+		return int64(read)
+	case *file.BooleanColumnChunkReader:
+		_, read, _ := r.ReadBatch(batch, p.ValuesOut.([]bool)[valuesRead:], defLevels, repLevels)
+		return int64(read)
+	case *file.FixedLenByteArrayColumnChunkReader:
+		_, read, _ := r.ReadBatch(batch, p.ValuesOut.([]parquet.FixedLenByteArray)[valuesRead:], defLevels, repLevels)
+		return int64(read)
+	default:
+		panic("unimplemented")
+	}
+}
+
+func Min(v1, v2 interface{}) interface{} {
+	switch n1 := v1.(type) {
+	case int32:
+		if n1 < v2.(int32) {
+			return n1
+		}
+		return v2
+	case int64:
+		if n1 < v2.(int64) {
+			return n1
+		}
+		return v2
+	case float32:
+		if n1 < v2.(float32) {
+			return n1
+		}
+		return v2
+	case float64:
+		if n1 < v2.(float64) {
+			return n1
+		}
+		return v2
+	}
+	panic("wtf")
+}
+
+func Max(v1, v2 interface{}) interface{} {
+	switch n1 := v1.(type) {
+	case int32:
+		if n1 < v2.(int32) {
+			return v2
+		}
+		return n1
+	case int64:
+		if n1 < v2.(int64) {
+			return v2
+		}
+		return n1
+	case float32:
+		if n1 < v2.(float32) {
+			return v2
+		}
+		return n1
+	case float64:
+		if n1 < v2.(float64) {
+			return v2
+		}
+		return n1
+	}
+	panic("wtf")
```

Comment:
is wtf common in Go panics?

Reply:
🤦  Nope, that's just something I forgot to fix with a more descriptive error message from when i was building this. Haha, i'll fix that.

Reply:
fixed